agent/api/task.go

```diff
@@ -847,6 +857,127 @@ func (task *Task) GetTaskENI() *ENI {
 	return task.ENI
 }
 
+// SetPullStartedAt sets the task pullstartedat timestamp and returns whether
+// this field was updated or not
+func (task *Task) SetPullStartedAt(timestamp time.Time) bool {
+	task.knownStatusTimeLock.Lock()
```

Comment:
Can we consolidate all of these locks into a single lock? It's hard to keep track of different locks for different methods. Let's have a single `sync.RWMutex` lock for the task. I understand that doing that is beyond the mandate of this PR. But, this is as good a time as any to make that change.